File: Problem2.py
#Time Complexity :O(n) 
#Space Complexity :O(1)
#Did this code successfully run on Leetcode : Yes
#Any problem you faced while coding this : No

class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        arr=[0]*26
        result=[]
        for i in range(len(s)):
            arr[ord(s[i])-ord('a')]=i

        start=0
        end=0
        for i in range(len(s)):
            c=s[i]
            end=max(end,arr[ord(s[i])-ord('a')])
            if i==end:
                result.append(end-start+1)
                start=end+1
        
        return result

        

File: problem2.java
class queueReconstructionByHeight {
    public int[][] reconstructQueue(int[][] people) {
        if(people == null || people.length == 0) return new int[][] {};
        
        Arrays.sort(people, (a, b) -> {
            if(a[0] == b[0]){
                return a[1] - b[1];
            }
            return b[0] - a[0];
        });
        
        List<int[]> result = new ArrayList();
        for(int[] person : people){
            result.add(person[1], person);
        }
        
        return result.toArray(new int[0][]);
    }
}
//time complexity O(n^2)
//space complexity O(1)